### PR TITLE
Add Kitti object detection dataset

### PR DESCRIPTION
**Pull request by sanskar107**
Added KITTI dataset.
~~Currently not inheriting from any base class. I am planning to add the base class once all the datasets are added and the structure is finalized.~~
@Prantl can you give some comments on whether the information from `get_data` and `get_attr` is sufficient ? as I have not included the image part yet.

### CODE REVIEW

File: ml3d/datasets/__init__.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 from .toronto3d import Toronto3D
 from .customdataset import Custom3D
 from .semantic3d import Semantic3D
-from .utils import *
+from .kitti import KITTI
+from . import utils
 __all__ = [
```
Comment (YilingQiao) [inline ml3d/datasets/__init__.py:13]: maybe we can also add KITTI to __all__

File: ml3d/datasets/kitti.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+from ..utils import Config, make_dir, DATASET
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(levelname)s - %(asctime)s - %(module)s - %(message)s',
+)
+log = logging.getLogger(__name__)
+
+
+class KITTI():
```
Comment (YilingQiao) [inline ml3d/datasets/kitti.py:18]: will it have a base class?

File: ml3d/datasets/kitti.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                 val_split=3712,
+                 **kwargs):
+        """
+        Initialize
+        Args:
+            dataset_path (str): path to the dataset
+            kwargs:
+        """
+
+        cfg = Config({
```
Comment (YilingQiao) [inline ml3d/datasets/kitti.py:37]: BaseDataset might be able to do this

File: ml3d/datasets/kitti.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+    def get_attr(self, idx):
+        pc_path = self.path_list[idx]
+        name = Path(pc_path).name.split('.')[0]
+
+        attr = {'name': name, 'path': pc_path, 'split': self.split}
+        return attr
+
+
+class Object3d(object):
```
Comment (YilingQiao) [inline ml3d/datasets/kitti.py:179]: can we add some docstrings to the new class?

**Review by Prantl — APPROVED** [review]
Looks good, I think it should fit like this. The image part is not needed at the moment.

File: ml3d/datasets/kitti.py (hunk 4 above)
Comment (sanskar107) [reply ml3d/datasets/kitti.py:180]: Done

File: ml3d/datasets/__init__.py (hunk 1 above)
Comment (sanskar107) [reply ml3d/datasets/__init__.py:13]: Done

File: ml3d/datasets/kitti.py (hunk 2 above)
Comment (sanskar107) [reply ml3d/datasets/kitti.py]: Done. The semseg base class works. If needed, we can add separate base class in future.

File: ml3d/datasets/kitti.py (hunk 3 above)
Comment (sanskar107) [reply ml3d/datasets/kitti.py]: Done

**Review by YilingQiao — APPROVED** [review]

**Pull request merged**